### PR TITLE
Add context propagation section in the cache extension guide

### PR DESCRIPTION
**Pull request by gwenneg**
Follow-up of #8227, related to #8166.

### CODE REVIEW

File: docs/src/main/asciidoc/cache.adoc [hunk 1]
```diff
@@ -331,6 +331,17 @@ quarkus.cache.caffeine."bar".maximum-size=1000 <2>
...
+== Context propagation
+
+This extension relies on non-blocking calls internally for cache values computations.
+By default, there's no context propagation between the calling thread (from your application) and a thread that performs such a computation.
+
+The context propagation can be enabled for this extension by simply including the `quarkus-smallrye-context-propagation` dependency into your project.
+
+If you see a `javax.enterprise.context.ContextNotActiveException` in your application log during a cache computation, then you probably need to enable the context propagation.
+
+You can find more information about context propagation in Quarkus in the link:context-propagation.html[dedicated guide].
```
Comment (gsmet) [inline docs/src/main/asciidoc/cache.adoc:343]:
```suggestion
You can find more information about context propagation in Quarkus in the link:context-propagation[dedicated guide].
```

File: docs/src/main/asciidoc/cache.adoc [hunk 2]
```diff
@@ -331,6 +331,17 @@ quarkus.cache.caffeine."bar".maximum-size=1000 <2>
 <1> The `foo` cache is being configured.
 <2> The `bar` cache is being configured.
 
+== Context propagation
+
+This extension relies on non-blocking calls internally for cache values computations.
+By default, there's no context propagation between the calling thread (from your application) and a thread that performs such a computation.
+
+The context propagation can be enabled for this extension by simply including the `quarkus-smallrye-context-propagation` dependency into your project.
```
Comment (gsmet) [inline docs/src/main/asciidoc/cache.adoc:339]:
```suggestion
The context propagation can be enabled for this extension by simply adding the `quarkus-smallrye-context-propagation` extension to your project.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Added some small suggestions.

**Comment by gwenneg** [comment]
Thanks for the review @gsmet! Your suggestions have been applied.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**